janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java

```diff
@@ -44,111 +26,90 @@
 import org.apache.tinkerpop.gremlin.structure.io.graphson.GraphSONMapper;
 import org.apache.tinkerpop.gremlin.structure.io.gryo.GryoMapper;
-import org.apache.velocity.Template;
-import org.apache.velocity.VelocityContext;
-import org.apache.velocity.app.Velocity;
-
-import com.google.common.base.Joiner;
-import com.google.common.collect.ImmutableMap;
 import org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
 
-public abstract class AbstractJanusGraphAssemblyIT {
+import java.net.Socket;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
 
-    protected static final String BUILD_DIR;
-    protected static final String EXPECT_DIR;
-    protected static final String ZIPFILE_PATH;
-    protected static final String ZIPFILE_EXTRACTED;
-    protected static final String ZIPFILE_FULL_PATH;
-    protected static final String ZIPFILE_FULL_EXTRACTED;
+import static org.apache.tinkerpop.gremlin.driver.ser.AbstractMessageSerializer.TOKEN_IO_REGISTRIES;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
 
-    static {
-        Properties props;
+public abstract class AbstractJanusGraphAssemblyIT extends JanusGraphAssemblyBaseIT {
 
-        try {
-            props = new Properties();
-            java.io.FileReader fr = new FileReader(Joiner.on(File.separator).join(new String[] { "target", "test-classes", "target.properties" }));
-            props.load(fr);
-            fr.close();
-        } catch (IOException e) {
-            throw new AssertionError(e);
-        }
+    protected abstract String getConfigPath();
 
-        BUILD_DIR    = props.getProperty("build.dir");
-        EXPECT_DIR   = props.getProperty("expect.dir");
-        ZIPFILE_PATH = props.getProperty("zipfile.path");
-        ZIPFILE_EXTRACTED = ZIPFILE_PATH.substring(0, ZIPFILE_PATH.length() - 4);
-        ZIPFILE_FULL_PATH = props.getProperty("zipfile-full.path");
-        ZIPFILE_FULL_EXTRACTED = ZIPFILE_FULL_PATH.substring(0, ZIPFILE_FULL_PATH.length() - 4);
+    protected abstract String getServerConfigPath();
 
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
-    }
+    protected abstract String getGraphName();
 
-    protected void testSimpleGremlinSession(String graphConfig, String graphToString, boolean full) throws Exception {
-        unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);
+    @ParameterizedTest
+    @ArgumentsSource(EnvironmentProvider.class)
```

Comment:
Should this be tagged with "FULL_TESTS"?

Reply:
I don't think so.

The tag full_tests should be just used in the provider to just run full tests under the condition that the tag full exists.

Reply:
If I am understanding correctly, according to the logic in EnvironmentProvider, the parameter `fullTest` will be always false unless it is tagged with `FULL_TESTS`. 

In the old code, however, I can see `single-vertex.expect.vm` is tested with `fullTest` being true & false.
```
    @Test
    public void testSimpleGremlinSession() throws Exception {
        testSimpleGremlinSession("conf/janusgraph-berkeleyje.properties", "berkeleyje", false);
    }

    @Test
    public void testSimpleGremlinSessionFull() throws Exception {
        testSimpleGremlinSession("conf/janusgraph-berkeleyje.properties", "berkeleyje", true);
    }
```

Thus, it seems to me that the new test is only covering the case when `fullTest == false`, while the old test covered both cases. Is this by intention?

Reply:
I add the handle to able to disable the full tests. It should execute same tests currently. I'll check it.

Reply:
I'll remove this logic with selecting tests and add it if need.